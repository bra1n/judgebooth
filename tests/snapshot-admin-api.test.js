const instance = require('./utils/client');
const dbUtils = require('./utils/db');

const doLogin = () => instance.get('/?action=test-auth');
const doLogout = () => instance.get('/?action=logout');

beforeEach(() => {
    return Promise.all([
        dbUtils.clearDatabase(),
        doLogout(),
    ]);
});

test.skip('snapshot /?action=auth without auth', async () => {
    try {
        const response = await instance.get('/?action=auth');
        fail();
    } catch (e) {
        expect(e.response.status).toBe(401);
        expect(e.response.data).toMatchSnapshot();
    }
});

test('snapshot /?action=auth with auth', async () => {
    await doLogin();
    const response = await instance.get('/?action=auth');
    expect(response.data).toMatchSnapshot();
});

test('snapshot /?action=admin-questions without auth', async () => {
    try {
        const response = await instance.get('/?action=admin-questions');
        fail();
    } catch (e) {
        expect(e.response.status).toBe(401);
        expect(e.response.data).toMatchSnapshot();
    }
});

test('snapshot /?action=admin-questions with auth', async () => {
    await doLogin();
    const response = await instance.get('/?action=admin-questions');
    expect(response.data).toMatchSnapshot();
});

test('snapshot /?action=admin-questions&page=2', async () => {
    await doLogin();
    const response = await instance.get('/?action=admin-questions&page=2');
    expect(response.data).toMatchSnapshot();
});


test('snapshot /?action=admin-question n.58 without auth', async () => {
    try {
        const response = await instance.get('/?action=admin-question&id=58');
        fail();
    } catch (e) {
        expect(e.response.status).toBe(401);
        expect(e.response.data).toMatchSnapshot();
    }
});

test('snapshot /?action=admin-question n. 58 with auth', async () => {
    await doLogin();
    const response = await instance.get('/?action=admin-question&id=58');
    expect(response.data).toMatchSnapshot();
});

test('snapshot /?action=admin-save n.58 without auth', async () => {
    try {
        const response = await instance.post('/?action=admin-save', {
            id: 58,
        });
        fail();
    } catch (e) {
        expect(e.response.status).toBe(401);
        expect(e.response.data).toMatchSnapshot();
    }
});

test('snapshot /?action=admin-save n. 58 with auth', async () => {
    await doLogin();
    const response = await instance.post('/?action=admin-save', {
        id: 58,
        live: 0,
    });
    const responseQ = await instance.get('/?action=admin-question&id=58');
    expect(responseQ.data.live).toBeFalsy();

    expect(response.data).toMatchSnapshot();
    responseQ.data.changedate = null;
    expect(responseQ.data).toMatchSnapshot();
});

test('snapshot /?action=admin-save new question with auth', async () => {
    await doLogin();
    const response = await instance.post('/?action=admin-save', {
        live: 1,
        author: 'John Doe',
        difficulty: 3,
        question: 'This is a temporary question',
        answer: 'SBA loves',
        cards: [
            {id: 146}, // Lightning Bolt
            {id: 9496}, // Tarmogoyf
        ]
    });
    const responseQ = await instance.get('/?action=admin-question&id=59');

    expect(response.data).toMatchSnapshot();
    responseQ.data.changedate = null;
    expect(responseQ.data).toMatchSnapshot();
});

test('snapshot /?action=admin-delete n.58 without auth', async () => {
    try {
        const response = await instance.post('/?action=admin-delete&id=58');
        fail();
    } catch (e) {
        expect(e.response.status).toBe(401);
        expect(e.response.data).toMatchSnapshot();
    }
});

test('snapshot /?action=admin-delete n.58 with auth', async () => {
    await doLogin();
    const response = await instance.post('/?action=admin-delete&id=58');
    expect(response.data).toMatchSnapshot();
});

test.skip('snapshot /?action=admin-suggest without auth', async () => {
    try {
        const response = await instance.get('/?action=admin-suggest&name=Black');
        fail();
    } catch (e) {
        expect(e.response.status).toBe(401);
        expect(e.response.data).toMatchSnapshot();
    }
});

test('snapshot /?action=admin-suggest with auth', async () => {
    await doLogin();
    const response = await instance.get('/?action=admin-suggest&name=Black');
    expect(response.data).toMatchSnapshot();
});

test('snapshot /?action=admin-translations without auth', async () => {
    try {
        const response = await instance.get('/?action=admin-translations&language=2');
        fail();
    } catch (e) {
        expect(e.response.status).toBe(401);
        expect(e.response.data).toMatchSnapshot();
    }
});

test('snapshot /?action=admin-translations with auth', async () => {
    await doLogin();
    const response = await instance.get('/?action=admin-translations&language=2');
    expect(response.data).toMatchSnapshot();
});

test('snapshot /?action=admin-translation n.58 lang=2 without auth', async () => {
    try {
        const response = await instance.get('/?action=admin-translation&id=58&lang=2');
        fail();
    } catch (e) {
        expect(e.response.status).toBe(401);
        expect(e.response.data).toMatchSnapshot();
    }
});

test('snapshot /?action=admin-translation n.58 lang=2 with auth', async () => {
    await doLogin();
    const response = await instance.get('/?action=admin-translation&id=58&lang=2');
    expect(response.data).toMatchSnapshot();
});

test('snapshot /?action=admin-translate n.58 lang=2 with auth', async () => {
    await doLogin();
    const response = await instance.post('/?action=admin-translate', {
        id: 58,
        language_id: 2,
        question: 'The question is changed',
        answer: 'The answer is changed',
    });
    expect(response.data).toEqual('success');
    const responseQ = await instance.get('/?action=admin-translation&id=58&lang=2');
    expect(responseQ.data.question_translated).toEqual('The question is changed');
    responseQ.data.changedate = null;
    responseQ.data.changedate_translated = null;

    expect(response.data).toMatchSnapshot();
    expect(responseQ.data).toMatchSnapshot();

});

test('snapshot /?action=admin-users without auth', async () => {
    try {
        const response = await instance.get('/?action=admin-users');
        fail();
    } catch (e) {
        expect(e.response.status).toBe(401);
        expect(e.response.data).toMatchSnapshot();
    }
});

test('snapshot /?action=admin-users with auth', async () => {
    await doLogin();
    const response = await instance.get('/?action=admin-users');
    expect(response.data).toMatchSnapshot();
});

test('snapshot /?action=admin-saveuser without auth', async () => {
    try {
        const response = await instance.post('/?action=admin-saveuser', {
            name: 'John Doe',
            email: 'john.doe@gmail.com',
            role: 'editor',
            languages: ['1', '2'],
        });
        fail();
    } catch (e) {
        expect(e.response.status).toBe(401);
        expect(e.response.data).toMatchSnapshot();
    }
});

test('snapshot /?action=admin-saveuser with auth', async () => {
    await doLogin();
    const response = await instance.post('/?action=admin-saveuser', {
        name: 'John Doe',
        email: 'john.doe@gmail.com',
        role: 'editor',
        languages: ['1', '2'],
    });
    expect(response.data).toMatchSnapshot();
});

test('snapshot /?action=admin-deleteuser without auth', async () => {
    try {
        const response = await instance.post('/?action=admin-deleteuser&email=boothadmin@gmail.com');
        fail();
    } catch (e) {
        expect(e.response.status).toBe(401);
        expect(e.response.data).toMatchSnapshot();
    }
});

test('snapshot /?action=admin-deleteuser with auth', async () => {
    await doLogin();
    const response = await instance.post('/?action=admin-deleteuser&email=boothadmin@gmail.com');
    expect(response.data).toMatchSnapshot();
});
