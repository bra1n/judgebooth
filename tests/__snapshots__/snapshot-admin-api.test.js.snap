// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot /?action=admin-delete n.58 with auth 1`] = `"success"`;

exports[`snapshot /?action=admin-delete n.58 without auth 1`] = `"unauthorized"`;

exports[`snapshot /?action=admin-deleteuser with auth 1`] = `"success"`;

exports[`snapshot /?action=admin-deleteuser without auth 1`] = `"unauthorized"`;

exports[`snapshot /?action=admin-question n. 58 with auth 1`] = `
Object {
  "answer": "0 Damage. Copy Enchantment can't copy anything that enters the battlefield at the same time as it does. Pandemonium does not have any ability, because of Humility and Opalescence.",
  "author": "",
  "cards": Array [
    Object {
      "id": 8122,
      "name": "Copy Enchantment",
    },
    Object {
      "id": 4202,
      "name": "Replenish",
    },
    Object {
      "id": 4187,
      "name": "Opalescence",
    },
    Object {
      "id": 3384,
      "name": "Pandemonium",
    },
    Object {
      "id": 2827,
      "name": "Humility",
    },
  ],
  "changedate": "2015-05-21 18:39:53",
  "difficulty": "2",
  "id": 58,
  "languages": Array [
    "1",
    "2",
    "4",
    "6",
    "7",
    "8",
    "9",
    "10",
    "11",
  ],
  "live": true,
  "question": "With Pandemonium, Humility, Copy Enchantment, and Opalescence in the graveyard, I cast Replenish. What's the highest amount of damage that I could deal?",
}
`;

exports[`snapshot /?action=admin-question n.58 without auth 1`] = `Array []`;

exports[`snapshot /?action=admin-questions with auth 1`] = `
Object {
  "pages": 1,
  "questions": Array [
    Object {
      "cards": Array [
        "Copy Enchantment",
        "Humility",
        "Opalescence",
        "Pandemonium",
        "Replenish",
      ],
      "difficulty": 2,
      "id": 58,
      "languages": Array [
        1,
        2,
        4,
        6,
        7,
        8,
        9,
        10,
        11,
      ],
      "live": true,
    },
  ],
}
`;

exports[`snapshot /?action=admin-questions without auth 1`] = `Array []`;

exports[`snapshot /?action=admin-questions&page=2 1`] = `
Object {
  "pages": 1,
  "questions": Array [],
}
`;

exports[`snapshot /?action=admin-save n. 58 with auth 1`] = `"success"`;

exports[`snapshot /?action=admin-save n. 58 with auth 2`] = `
Object {
  "answer": "0 Damage. Copy Enchantment can't copy anything that enters the battlefield at the same time as it does. Pandemonium does not have any ability, because of Humility and Opalescence.",
  "author": "",
  "cards": Array [
    Object {
      "id": 8122,
      "name": "Copy Enchantment",
    },
    Object {
      "id": 4202,
      "name": "Replenish",
    },
    Object {
      "id": 4187,
      "name": "Opalescence",
    },
    Object {
      "id": 3384,
      "name": "Pandemonium",
    },
    Object {
      "id": 2827,
      "name": "Humility",
    },
  ],
  "changedate": null,
  "difficulty": "2",
  "id": 58,
  "languages": Array [
    "1",
    "2",
    "4",
    "6",
    "7",
    "8",
    "9",
    "10",
    "11",
  ],
  "live": false,
  "question": "With Pandemonium, Humility, Copy Enchantment, and Opalescence in the graveyard, I cast Replenish. What's the highest amount of damage that I could deal?",
}
`;

exports[`snapshot /?action=admin-save n.58 without auth 1`] = `"unauthorized"`;

exports[`snapshot /?action=admin-save new question with auth 1`] = `"success"`;

exports[`snapshot /?action=admin-save new question with auth 2`] = `
Object {
  "answer": "SBA loves",
  "author": "John Doe",
  "cards": Array [
    Object {
      "id": 146,
      "name": "Lightning Bolt",
    },
    Object {
      "id": 9496,
      "name": "Tarmogoyf",
    },
  ],
  "changedate": null,
  "difficulty": "3",
  "id": 59,
  "languages": Array [
    "1",
  ],
  "live": false,
  "question": "This is a temporary question",
}
`;

exports[`snapshot /?action=admin-saveuser with auth 1`] = `"success"`;

exports[`snapshot /?action=admin-saveuser without auth 1`] = `"unauthorized"`;

exports[`snapshot /?action=admin-suggest with auth 1`] = `
Array [
  Object {
    "full_name": null,
    "id": 1454,
    "name": "Black Carriage",
  },
  Object {
    "full_name": null,
    "id": 12613,
    "name": "Black Cat",
  },
  Object {
    "full_name": null,
    "id": 18,
    "name": "Black Knight",
  },
  Object {
    "full_name": null,
    "id": 19,
    "name": "Black Lotus",
  },
  Object {
    "full_name": null,
    "id": 486,
    "name": "Black Mana Battery",
  },
  Object {
    "full_name": null,
    "id": 4286,
    "name": "Black Market",
  },
  Object {
    "full_name": null,
    "id": 14080,
    "name": "Black Oak of Odunos",
  },
  Object {
    "full_name": null,
    "id": 9567,
    "name": "Black Poplar Shaman",
  },
  Object {
    "full_name": null,
    "id": 1116,
    "name": "Black Scarab",
  },
  Object {
    "full_name": null,
    "id": 9513,
    "name": "Black Sun's Zenith",
  },
]
`;

exports[`snapshot /?action=admin-suggest without auth 1`] = `Array []`;

exports[`snapshot /?action=admin-translate n.58 lang=2 with auth 1`] = `"success"`;

exports[`snapshot /?action=admin-translate n.58 lang=2 with auth 2`] = `
Object {
  "answer": "0 Damage. Copy Enchantment can't copy anything that enters the battlefield at the same time as it does. Pandemonium does not have any ability, because of Humility and Opalescence.",
  "answer_translated": "The answer is changed",
  "author": "",
  "cardids": Array [
    "111640",
    "0",
    "0",
    "131326",
    "0",
  ],
  "cards": Array [
    "Verzauberungskopie",
    "Replenish",
    "Opalescence",
    "Pandämonium",
    "Humility",
  ],
  "changedate": null,
  "changedate_translated": null,
  "difficulty": 2,
  "id": 58,
  "language_id": 2,
  "live": true,
  "question": "With Pandemonium, Humility, Copy Enchantment, and Opalescence in the graveyard, I cast Replenish. What's the highest amount of damage that I could deal?",
  "question_translated": "The question is changed",
}
`;

exports[`snapshot /?action=admin-translation n.58 lang=2 with auth 1`] = `
Object {
  "answer": "0 Damage. Copy Enchantment can't copy anything that enters the battlefield at the same time as it does. Pandemonium does not have any ability, because of Humility and Opalescence.",
  "answer_translated": "0 Schaden. Die Verzauberungskopie kann nichts kopieren, das gleichzeitig mit ihr das Schlachtfeld betritt. Pandämonium hat keine Fähigkeiten, da sie von Opalessenz und Humility betroffen ist.",
  "author": "",
  "cardids": Array [
    "111640",
    "0",
    "0",
    "131326",
    "0",
  ],
  "cards": Array [
    "Verzauberungskopie",
    "Replenish",
    "Opalescence",
    "Pandämonium",
    "Humility",
  ],
  "changedate": "2015-05-21 18:39:53",
  "changedate_translated": "2015-06-04 12:07:33",
  "difficulty": 2,
  "id": 58,
  "language_id": 2,
  "live": true,
  "question": "With Pandemonium, Humility, Copy Enchantment, and Opalescence in the graveyard, I cast Replenish. What's the highest amount of damage that I could deal?",
  "question_translated": "Während ich Pandämonium, Humility, Verzauberungskopie und Opalessenz auf meinem Firedhof habe, wirke ich Replenish. Wie hoch ist die größte Anzahl an Schaden, die ich zufügen könnte?",
}
`;

exports[`snapshot /?action=admin-translation n.58 lang=2 without auth 1`] = `Array []`;

exports[`snapshot /?action=admin-translations with auth 1`] = `
Array [
  Object {
    "cards": Array [
      "Copy Enchantment",
      "Replenish",
      "Opalescence",
      "Pandemonium",
      "Humility",
    ],
    "live": true,
    "question_id": 58,
    "status": "untranslated",
  },
]
`;

exports[`snapshot /?action=admin-translations without auth 1`] = `Array []`;

exports[`snapshot /?action=admin-users with auth 1`] = `
Array [
  Object {
    "email": "boothadmin@gmail.com",
    "name": "Admin Istrator",
    "role": "admin",
  },
]
`;

exports[`snapshot /?action=admin-users without auth 1`] = `Array []`;

exports[`snapshot /?action=auth with auth 1`] = `
Object {
  "email": "boothadmin@gmail.com",
  "languages": Array [],
  "name": "Admin Istrator",
  "role": "admin",
}
`;

exports[`snapshot /?action=auth without auth 1`] = `
Object {
  "error": "No account found",
}
`;
